### PR TITLE
Implement `std::string` and `std::wstring`

### PR DESCRIPTION
**Pull request by Mr-Auto**
Never seen `std::wstring` in the game itself, but it was easy enough to make just in case

### CODE REVIEW

File: src/QtHelpers/TreeViewMemoryFields.cpp [hunk 1]
```diff
@@ -1491,6 +1492,84 @@ void S2Plugin::TreeViewMemoryFields::updateValueForField(const MemoryField& fiel
...
+            {
+                comparisonBuffer = comparison.get_string();
+            }
+            itemComparisonValue->setData(QString(comparisonBuffer.get()), Qt::DisplayRole);
+            auto hexComparisonValue = QString::asprintf("<font color='blue'><u>0x%016llX</u></font>", comparisonValue);
+            itemComparisonValueHex->setData(hexComparisonValue, Qt::DisplayRole);
+            itemComparisonValue->setBackground(value != comparisonValue ? comparisonDifferenceColor : Qt::transparent);
+            itemComparisonValueHex->setBackground(value != comparisonValue ? comparisonDifferenceColor : Qt::transparent);
+
+            if (shouldUpdateChildren)
```
Comment (zappatic) [inline src/QtHelpers/TreeViewMemoryFields.cpp:1525]: Is this applicable for a std::(w)string ?

Comment (Mr-Auto) [reply src/QtHelpers/TreeViewMemoryFields.cpp:1525]: probably not, unless it's `std::string*` or if you compare it with itself

Comment (Mr-Auto) [reply src/QtHelpers/TreeViewMemoryFields.cpp:1525]: will change that later

Comment (Mr-Auto) [reply src/QtHelpers/TreeViewMemoryFields.cpp:1525]: Done, i left the hex comparison as it's weird to show different values but color them as the same

**Pull request merged**